### PR TITLE
build: make build.sh work locally after the CI log cleanup

### PR DESCRIPTION
**Pull request by IgorMinar**

### CODE REVIEW

File: scripts/ci-lite/env.sh [hunk 1]
```diff
@@ -27,13 +27,10 @@ function  setEnvVar() {
 }
 
 
-# strip leading "./"
-currentFileName=${0#./}
-currentWorkingDirectory=`pwd`
-#cd ${currentWorkingDirectory%currentFileName}
+readonly currentDir=$(cd $(dirname $0); pwd)
```
Comment (ocombe) [inline scripts/ci-lite/env.sh]: this breaks on travis: `cd: ./scripts/ci: No such file or directory`

File: scripts/ci-lite/_travis_fold.sh [hunk 2]
```diff
@@ -13,8 +13,10 @@ function travisFoldStart() {
...
   echo ""
-  echo "travis_fold:start:${sanitizedFoldName}"
-  echo "travis_time:start:${sanitizedFoldName}"
+  if [[ ${TRAVIS:-} ]]; then
+    echo "travis_fold:start:${sanitizedFoldName}"
+    echo "travis_time:start:${sanitizedFoldName}"
+  fi
   local enterArrow="===>  ${foldName}  ==>==>==>==>==>==>==>==>==>==>==>==>==>==>==>==>==>==>==>==>==>==>==>==>==>==>==>==>==>"
   # keep all messages consistently wide 80chars regardless of the foldName
   echo ${enterArrow:0:100}
```
Comment (chuckjaz) [inline scripts/ci-lite/_travis_fold.sh]: Consider only enabling "+x" when building on travis and using a --verbose switch.

The current build is very chatty and locally I rarely will care about the verbose output.

**Review by chuckjaz — COMMENTED** [review]
Consider a follow-on PR to reduce the verbosity of local builds.

File: scripts/ci-lite/env.sh (hunk 1 above)
Comment (chuckjaz) [inline scripts/ci-lite/env.sh]: Consider changing this to `readonly currentDir=$(cd $(dirname $0)/../ci-lite; pwd)`

**Comment by IgorMinar** [comment]
@chuckjaz @ocombe I resolved all your comments and in fact made a much bigger change because the way we were propagating env variables was flawed.

I've merged this in to unblock everyone but can you please take one more look to make sure I didn't miss anything. I'll be happy to craft a follow up PR to fix it. Please review individual commits because otherwise you'll see a big removal and addition of files rather than move + edit.

File: scripts/ci/env.sh [hunk 3]
```diff
@@ -1,18 +1,12 @@
...
-# set wasBashSetXOn using the current "set -x" mode state, so that we can restore it at the end
-[[ "${-//[^x]/}" = "x" ]] && wasBashSetXOn=1 || wasBashSetXOn=0
+# because this script is being source-ed via .travis.yaml,
+# we need to restore the original options so that that we don't interfere with
+# travis' internals
+readonly ORIGINAL_SHELL_OPTIONS=$(set +o)
 
 # this script is extra noisy and used in many places during the build so we suppress the trace with +x to reduce the noise
-set +x -u -e -o pipefail
+set -u -e -o pipefail
```
Comment (gkalpak) [inline scripts/ci/env.sh]: Why remove the `+x`? Is it off by default? (Also the comment above doesn't apply any more.)

File: scripts/ci/install.sh [hunk 4]
```diff
@@ -43,33 +49,41 @@ fi
 if [[ ${TRAVIS} && (${CI_MODE} == "aio" || ${CI_MODE} == "docs_test") ]]; then
   # angular.io: Install all yarn dependencies according to angular.io/yarn.lock
   travisFoldStart "yarn-install.aio"
-    cd "`dirname $0`/../../aio"
-    yarn install
-    cd -
+    (
+      printenv
```
Comment (gkalpak) [inline scripts/ci/install.sh]: 😱 Is this a debugging left-over? We shouldn't print `NGBUILDS_IO_KEY` (once it is #14888 is merged).

**Comment by ocombe** [comment]
it works for me locally with your changes (for my usage, which is build and offline_compiler_test), but it seems to be failing on travis

**Pull request merged**